system/jlib/jevent.cpp

```diff
@@ -91,28 +91,38 @@ struct EventInformation
     bool isMeta;                // Is this meta data (see description below) rather than an event
     EventType pairedEvent;      // could be used to automaticaly link start and finish events?
     EventContext context;       // grouping related events
+    EventAttr attributes[EvAttrMax];     // EvAttrNone-terminated array of attributes
 };
 
 
-#define DEFINE_EVENT(event, ctx) { Event##event, #event, false, EventNone, ctx }
-#define DEFINE_META(meta, ctx) { Meta##meta, #meta, true, EventNone, ctx }
+#define DEFINE_EVENT(event, ctx, attrs) { Event##event, #event, false, EventNone, ctx, attrs }
+#define DEFINE_META(meta, ctx, attrs) { Meta##meta, #meta, true, EventNone, ctx, attrs }
+#define ATTR_HEADER           { EvAttrEventTimestamp, EvAttrEventTraceId, EvAttrEventThreadId, EvAttrEventStackTrace
+#define ATTR_FOOTER           EvAttrNone }
+#define INDEX_HEADER          ATTR_HEADER, EvAttrFileId, EvAttrFileOffset, EvAttrNodeKind
+#define INDEXLOOKUP_ATTRS     INDEX_HEADER, EvAttrInCache, EvAttrExpandedSize, ATTR_FOOTER
+#define INDEXLOAD_ATTRS       INDEX_HEADER, EvAttrExpandedSize, EvAttrElapsedTime, EvAttrReadTime, ATTR_FOOTER
+#define INDEXEVICTION_ATTRS   INDEX_HEADER, EvAttrExpandedSize, ATTR_FOOTER
+#define DALI_ATTRS            ATTR_HEADER, EvAttrPath, EvAttrConnectId, EvAttrElapsedTime, EvAttrDataSize, ATTR_FOOTER
+#define FILEINFORMATION_ATTRS ATTR_HEADER, EvAttrFileId, EvAttrPath, ATTR_FOOTER
+#define RECORDINGACTIVE_ATTRS ATTR_HEADER, EvAttrEnabled, ATTR_FOOTER
 
 static constexpr EventInformation eventInformation[] {
-    DEFINE_EVENT(None, EventCtxMax),
-    DEFINE_EVENT(IndexLookup, EventCtxIndex),
-    DEFINE_EVENT(IndexLoad, EventCtxIndex),
-    DEFINE_EVENT(IndexEviction, EventCtxIndex),
-    DEFINE_EVENT(DaliChangeMode, EventCtxDali),
-    DEFINE_EVENT(DaliCommit, EventCtxDali),
-    DEFINE_EVENT(DaliConnect, EventCtxDali),
-    DEFINE_EVENT(DaliEnsureLocal, EventCtxDali),
-    DEFINE_EVENT(DaliGet, EventCtxDali),
-    DEFINE_EVENT(DaliGetChildren, EventCtxDali),
-    DEFINE_EVENT(DaliGetChildrenFor, EventCtxDali),
-    DEFINE_EVENT(DaliGetElements, EventCtxDali),
-    DEFINE_EVENT(DaliSubscribe, EventCtxDali),
-    DEFINE_META(FileInformation, EventCtxIndex),
-    DEFINE_EVENT(RecordingActive, EventCtxOther),
+    DEFINE_EVENT(None, EventCtxMax, { EvAttrNone } ),
+    DEFINE_EVENT(IndexLookup, EventCtxIndex, INDEXLOOKUP_ATTRS ),
```

Comment:
Style.  I would always include the { } in this list here, so the macros only expand to a list.